### PR TITLE
Settle funding fees

### PR DESCRIPTION
**Pull request by gmxer**
Related PR https://github.com/gmx-io/gmx-subgraph/pull/21

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204804284562508

### CODE REVIEW

File: src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  const matchedOrderStatuses = useMemo(
+    () => matchedOrderStatusKeys.map((key) => allOrderStatuses[key]),
+    [allOrderStatuses, matchedOrderStatusKeys]
+  );
+
+  const [keyByOrder, orderByKey] = useMemo(() => {
+    const map1 = new Map<PendingOrderData, string>();
+    const map2 = new Map<string, PendingOrderData>();
+    orders.forEach((order) => {
+      const key = getPendingOrderKey(order);
```
Comment (gdev8317) [inline src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx:37]: just to confirm, why not use `order.key` itself?

Comment (gdev8317) [reply src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx:37]: i guess in this case we may not need to have these two maps

File: src/pages/SyntheticsPage/SyntheticsPage.tsx [hunk 2]
```diff
@@ -463,7 +489,7 @@ export function SyntheticsPage(p: Props) {
                 [ListSection.Positions]: t`Positions${positionsCount ? ` (${positionsCount})` : ""}`,
                 [ListSection.Orders]: renderOrdersTabTitle(),
                 [ListSection.Trades]: t`Trades`,
-                [ListSection.Claims]: t`Claims`,
+                [ListSection.Claims]: hasClaimables ? t`Claims (1)` : t`Claims`,
```
Comment (gdev8317) [inline src/pages/SyntheticsPage/SyntheticsPage.tsx:492]: is it okay that `1` is hardcoded?

File: src/components/Synthetics/SettleAccruedFundingFeeModal/SettleAccruedFundingFeeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        return {
+          account,
+          marketAddress: position.marketAddress,
+          initialCollateralAddress: position.collateralTokenAddress,
+          initialCollateralDeltaAmount: BigNumber.from(1), // FIXME ?
+          receiveTokenAddress: position.collateralToken.address,
+          swapPath: [],
+          sizeDeltaUsd: BigNumber.from(0),
+          sizeDeltaInTokens: BigNumber.from(0),
+          acceptablePrice: markPrice!,
```
[the remark this answers is not in the document]
Comment (gmxer) [reply src/components/Synthetics/SettleAccruedFundingFeeModal/SettleAccruedFundingFeeModal.tsx:125]: out-of-bounds with `.pow(256)`, least number that works is `BigNumber.from(10).pow(95)`

File: src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx (hunk 1 above)
Comment (gmxer) [reply src/components/Synthetics/StatusNotification/FeesSettlementStatusNotification.tsx:37]: 1. that is only way to connect `order` with `orderStatus` (`PendingOrderData` doesn't contain `key`)
2. `getPendingOrderKey(order) != order.key`

File: src/pages/SyntheticsPage/SyntheticsPage.tsx (hunk 2 above)
Comment (gmxer) [reply src/pages/SyntheticsPage/SyntheticsPage.tsx:492]: https://app.asana.com/0/home/1205451119040156/1204804284562508
>3. If the user has any Pending Claimable Fees then Claims tab will have an (1) next to it

yes, that's fine

File: src/components/Synthetics/SettleAccruedFundingFeeModal/SettleAccruedFundingFeeModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gdev8317) [reply src/components/Synthetics/SettleAccruedFundingFeeModal/SettleAccruedFundingFeeModal.tsx]: interesting, it works for me on node side

```
<img width="552" alt="image" src="https://github.com/gmx-io/gmx-interface/assets/95585424/140aef33-7ab9-4887-ac6b-c0b6afa2b7ba">

```

**Review by gdev8317 — APPROVED** [review]

**Pull request merged**